src/wp-includes/link-template.php

```diff
@@ -1469,10 +1469,15 @@ function get_edit_post_link( $post = 0, $context = 'display' ) {
 		return;
 	}
 
-	if ( $post_type_object->_edit_link ) {
-		$link = admin_url( sprintf( $post_type_object->_edit_link . $action, $post->ID ) );
+	if ( 'wp_template' === $post->post_type || 'wp_template_part' === $post->post_type ) {
+		$slug = urlencode( get_stylesheet() . '//' . $post->post_name );
+		$link = admin_url( sprintf( $post_type_object->_edit_link, $post->post_type, $slug ) );
```

Comment:
Do we need to encode `&` based on the value of `$context` similar to what is happening for post edit links above?

Reply:
Also I think we'd need to update the annotations, e.g.,


```
 * Can be used within the WordPress loop or outside of it. Can be used with
 * pages, posts, attachments, revisions, templates and template parts.
 *
 * @since 2.3.0
 * @since 6.3.0 Updated to handle `_edit_link` properties for templates and template parts.
 ...
```

Reply:
I don't understand the bit about `$context`?

Reply:
I don't know why it's there either 😄 Maybe a vestigial feature that's not really relevant anymore?

I just noticed that when building post links [the ampersand is encoded](https://github.com/WordPress/wordpress-develop/blob/6c2d9c3ee6877984083c05c897c08631be1d6e0d/src/wp-includes/link-template.php#L1457) depending on the value of the argument `$context`


```php
function get_edit_post_link( $post = 0, $context = 'display' ) {
	$post = get_post( $post );

	if ( ! $post ) {
		return;
	}

	if ( 'revision' === $post->post_type ) {
		$action = '';
	} elseif ( 'display' === $context ) {
		$action = '&amp;action=edit';
	} else {
		$action = '&action=edit';
	}
....
```

Reply:
Aaah yes! Well the test seems to take that into acount so 🤷🏻